packages/babel-helper-module-transforms/src/normalize-and-load-metadata.ts

```diff
@@ -105,16 +133,24 @@ export default function normalizeModuleAndLoadMetadata(
       metadata.name = metadata.importsNamespace.values().next().value;
     }
 
-    if (noInterop) metadata.interop = "none";
-    else if (esNamespaceOnly) {
+    const resolvedInterop = resolveImportInterop(
+      importInterop,
+      metadata.source,
+    );
+
+    if (resolvedInterop === "none") {
+      metadata.interop = "none";
+    } else if (resolvedInterop === "node" && metadata.interop === "namespace") {
```

Comment:
What is `metadata.interop`?

Reply:
It tells what interop function we need for the import:
- default -> default
- named -> none
- default+named -> namespace
- namespace -> namespace

Then in this function we specialize between "babel interop" and "node interop".